src/pages/resultsView/plots/PlotsTabUtils.tsx

```diff
@@ -1934,158 +1963,120 @@ export function makeScatterPlotPointAppearance(
         ? oncoprintMutationTypeToAppearanceDrivers
         : oncoprintMutationTypeToAppearanceDefault;
 
-    switch (coloringType) {
-        case ColoringType.ClinicalData:
-            if (!coloringMenuSelectedOption || !clinicalDataCache) {
-                break;
-            }
+    let ret = null;
 
-            const data = clinicalDataCache.get(
-                coloringMenuSelectedOption.info.clinicalAttribute!
+    if (ColoringType.ClinicalData in coloringType) {
+        // clinical data precludes all others
+        ret = makeScatterPlotPointAppearance_Clinical(
+            coloringMenuSelectedOption,
+            clinicalDataCache,
+            coloringLogScale
+        );
+    } else {
+        let getMutApp: any = () => ({});
+        let getCnaApp: any = () => ({});
+        let getLimitValueApp: any = () => ({});
+        if (
+            ColoringType.MutationType in coloringType &&
+            mutationDataExists.isComplete &&
+            mutationDataExists.result
+        ) {
+            getMutApp = (d: IPlotSampleData) =>
+                getMutationTypeAppearance(d, oncoprintMutationTypeToAppearance);
+        }
+        if (
+            (ColoringType.CopyNumber in coloringType &&
+                cnaDataExists.isComplete &&
```

Comment:
i see this isn't new, but it seems weird to me that we would be checking for loading state inside this function.  that suggests that we would be rendering something ("appearance") prior to the completion of data on which the appearance depends?

Reply:
I agree, this did strike me as weird too. I decided just to go with the flow because Im overwhelmed by how large and complicated this tab is..

Reply:
@adamabeshouse this one here